src/React/Widgets/SelectionEditorWidget/partition/index.js

```diff
@@ -5,19 +5,59 @@ import DividerRender    from './DividerRender';
 import SelectionBuilder from '../../../../Common/Misc/SelectionBuilder';
 
 export default function partitionSelection(props) {
-  const dividers = props.selection.partition.dividers;
-  const fieldName = props.selection.partition.variable;
+  function clampDividerUncertainty(dividers, index, inMaxUncertainty) {
```

Comment:
Can you put those functions outside the main one so they don't get created as closure?
You will need more args, but that should be better.